### PR TITLE
Adds event listener on target for blur events

### PR DESCRIPTION
**Pull request by chrisaugustsson**
Related to https://github.com/tamagui/tamagui/issues/2223

Using input inside FocusScope with trapped focus, the input gets reselected on blur events.

This PR adds a eventlistener for blur events, and sets focus to the container if blur events are triggered.

### CODE REVIEW

**Comment by natew** [comment]
We'd have to do this differently, but the idea is right. You need to handle/dispose that event listener.

**Comment by chrisaugustsson** [comment]
> We'd have to do this differently, but the idea is right. You need to handle/dispose that event listener.

What do you think about using a abort controller to remove the event listener?

**Comment by natew** [comment]
this looks much better

File: packages/focus-scope/src/FocusScope.tsx [hunk 1]
```diff
@@ -66,26 +66,37 @@ export function useFocusScope(
   React.useEffect(() => {
     if (!enabled) return
     if (!trapped) return
+    const controller = new AbortController;
```
Comment (natew) [inline packages/focus-scope/src/FocusScope.tsx:69]:
```suggestion
    const controller = new AbortController();
```

??

**Comment by natew** [comment]
merged in another pr, ty

**Pull request closed without merging**